### PR TITLE
Cleanup principles

### PR DESCRIPTION
**Pull request by duglin**
I didn't really change much, just moved somes stuff around and expanded
a little more in number 5.

I moved all of the physical shipping container stuff to just the into
because while its a cute analogy, repeating it over and over just got
in the way of the real points - and by number 5 we ended up having more text
about shipping containers than our containers - which was just weird.

Signed-off-by: Doug Davis <dug@us.ibm.com>

### CODE REVIEW

File: principles.md [hunk 1]
```diff
@@ -1,51 +1,71 @@
...
 Define a unit of software delivery called a Standard Container.
-The goal of a Standard Container is to encapsulate a software component and all its dependencies in a format that is self-describing and portable, so that any compliant runtime can run it without extra dependencies, regardless of the underlying machine and the contents of the container.
-
-The specification for Standard Containers is straightforward.
-It mostly defines 1) a file format, 2) a set of standard operations, and 3) an execution environment.
-
-A great analogy for this is the shipping container.
-Just like how Standard Containers are a fundamental unit of software delivery, shipping containers are a fundamental unit of physical delivery.
+The goal of a Standard Container is to encapsulate a software component and
+all its dependencies in a format that is self-describing and portable, so that
```
Comment (wking) [inline principles.md]: Markdown policy is [one sentence per line][1] (which is where the long lines from the original came from ;).

[1]: https://github.com/opencontainers/specs/blob/v0.1.1/README.md#markdown-style

Comment (duglin) [inline principles.md]: I can switch it if people want, I'm just used to the docker practice of trying not to span 80 chars.  No biggie.

**Comment by wking** [comment]
On Fri, Oct 02, 2015 at 06:48:28AM -0700, Doug Davis wrote:
> I moved all of the physical shipping container stuff to just the
> [intro] because while its a cute analogy, repeating it over and over
> just got in the way of the real points…

+1 to this.

> - and by number 5 we ended up having more text about shipping
> containers than our containers - which was just weird.

I still think it would be nice to have more clarity about what
“Industrial-grade delivery” is about (both for the container ecosystem
as a whole, and for this spec in particular), but that's been
contentious in the past [1,2], so it's probably best to leave it out
of this PR ;).

[1]: https://groups.google.com/a/opencontainers.org/forum/#!topic/dev/uo11avcWlQQ
[2]: https://groups.google.com/a/opencontainers.org/forum/#!topic/dev/OqnUp4jOacs

**Comment by vbatts** [comment]
@duglin across the whole doc, we do formatting of a sentence on each line (no line wraps in a sentence).

**Comment by duglin** [comment]
okie dokie - one sentence per line.

File: principles.md [hunk 2]
```diff
@@ -3,49 +3,42 @@
...
 The goal of a Standard Container is to encapsulate a software component and all its dependencies in a format that is self-describing and portable, so that any compliant runtime can run it without extra dependencies, regardless of the underlying machine and the contents of the container.
 
-The specification for Standard Containers is straightforward.
-It mostly defines 1) a file format, 2) a set of standard operations, and 3) an execution environment.
+The specification for Standard Containers defines:
 
-A great analogy for this is the shipping container.
-Just like how Standard Containers are a fundamental unit of software delivery, shipping containers are a fundamental unit of physical delivery.
+1. configuration file formats
+2. a set of standard operations
```
Comment (vbatts) [inline principles.md]: wondered if you were going to touch this, as it bothered you some weeks ago

Comment (duglin) [inline principles.md]: well, this round is more about clean-up - with just a touch of clarity.  The entire 'standard operations' thing is still a bit of a mystery to me since I don't know how you define an operation w/o defining something like an interface or API  :-)

Comment (vbatts) [inline principles.md]: fair

File: principles.md [hunk 3]
```diff
@@ -3,49 +3,42 @@
...
+
+A great analogy for this is the physical shipping container used by the transportation industry.
+Shipping containers are a fundamental unit of delivery, they can be lifted, stacked, locked, loaded, unloaded and labelled.
+Irrespective of their contents, by standardizing the container itself it allowed for a consistent, more streamlined and efficient set of processes to be defined.
+For software Standard Containers offer similar functionality by being the fundamental, standardized, unit of delivery for a software package.
 
 ## 1. Standard operations
 
-Just like shipping containers, Standard Containers define a set of STANDARD OPERATIONS.
-Shipping containers can be lifted, stacked, locked, loaded, unloaded and labelled.
```
Comment (vbatts) [inline principles.md]: I just read this thinking that we had labels/annotations but then realized that we haven't added such yet :-\
I see https://github.com/opencontainers/specs/issues/108 and think we should move forward on such

**Comment by vbatts** [comment]
all-in-all, I'm not opposed. The traditional container is referenced early, but then the similarities are expressed in the sections. For verbosity sake, every 1:1 analogy is not needed.

**Comment by mrunalp** [comment]
Looks good overall. @crosbymichael @philips PTAL

File: principles.md [hunk 4]
```diff
@@ -3,49 +3,42 @@
...
+Standard Containers are CONTENT-AGNOSTIC: all standard operations have the same effect regardless of the contents.
+They are started in the same way whether they contain a postgres database, a php application with its dependencies and application server, or Java build artifacts.
 
 ## 3. Infrastructure-agnostic
 
-Both types of containers are INFRASTRUCTURE-AGNOSTIC: they can be transported to thousands of facilities around the world, and manipulated by a wide variety of equipment.
-A shipping container can be packed in a factory in Ukraine, transported by truck to the nearest routing center, stacked onto a train, loaded into a German boat by an Australian-built crane, stored in a warehouse at a US facility, etc.
-Similarly, a standard container can be bundled on my laptop, uploaded to S3, downloaded, run and snapshotted by a build server at Equinix in Virginia, uploaded to 10 staging servers in a home-made Openstack cluster, then sent to 30 production instances across 3 EC2 regions.
+Standard Containers are INFRASTRUCTURE-AGNOSTIC: they can be run in any OCI compliant infrastructure.
```
Comment (crosbymichael) [inline principles.md]: `s/OCI compliant infrastructure./OCI supported infrastructure.`  

This kinda makes it sound like we are going to have compliance testing for hardware or something like that.  The burden would be on us or the runtimes to support the infrastructure, not a burden on the infrastructure providers.

Comment (duglin) [inline principles.md]: done

File: principles.md [hunk 5]
```diff
@@ -3,49 +3,42 @@
...
+Standard Containers are INFRASTRUCTURE-AGNOSTIC: they can be run in any OCI supported infrastructure.
+For example, a standard container can be bundled on a laptop, uploaded to S3, downloaded, run and snapshotted by a build server at Equinix in Virginia, uploaded to 10 staging servers in a home-made OpenStack cluster, then sent to 30 production instances across 3 EC2 regions.
 
 ## 4. Designed for automation
 
-Because they offer the same standard operations regardless of content and infrastructure, Standard Containers, just like their physical counterparts, are extremely well-suited for automation.
-In fact, you could say automation is their secret weapon.
+Standard Containers are DESIGNED FOR AUTOMATION: because they offer the same standard operations regardless of content and infrastructure, Standard Containers, are extremely well-suited for automation.  In fact, you could say automation is their secret weapon.
```
Comment (vbatts) [inline principles.md]: one sentence per line, please

File: principles.md [hunk 6]
```diff
@@ -3,49 +3,42 @@
...
+Shipping containers are a fundamental unit of delivery, they can be lifted, stacked, locked, loaded, unloaded and labelled.
+Irrespective of their contents, by standardizing the container itself it allowed for a consistent, more streamlined and efficient set of processes to be defined.
+For software Standard Containers offer similar functionality by being the fundamental, standardized, unit of delivery for a software package.
 
 ## 1. Standard operations
 
-Just like shipping containers, Standard Containers define a set of STANDARD OPERATIONS.
-Shipping containers can be lifted, stacked, locked, loaded, unloaded and labelled.
-Similarly, Standard Containers can be created, started, and stopped using standard container tools (what this spec is about); copied and snapshotted using standard filesystem tools; and downloaded and uploaded using standard network tools.
+Standard Containers define a set of STANDARD OPERATIONS.  They can be created, started, and stopped using standard container tools; copied and snapshotted using standard filesystem tools; and downloaded and uploaded using standard network tools.
```
Comment (vbatts) [inline principles.md]: one sentence per line

**Comment by vbatts** [comment]
couple of nits, but LGTM

**Comment by duglin** [comment]
fixed the multiple sentences per line issues that @vbatts found.

**Comment by vbatts** [comment]
LGTM

File: principles.md [hunk 7]
```diff
@@ -3,49 +3,44 @@
...
 
 ## 3. Infrastructure-agnostic
 
-Both types of containers are INFRASTRUCTURE-AGNOSTIC: they can be transported to thousands of facilities around the world, and manipulated by a wide variety of equipment.
-A shipping container can be packed in a factory in Ukraine, transported by truck to the nearest routing center, stacked onto a train, loaded into a German boat by an Australian-built crane, stored in a warehouse at a US facility, etc.
-Similarly, a standard container can be bundled on my laptop, uploaded to S3, downloaded, run and snapshotted by a build server at Equinix in Virginia, uploaded to 10 staging servers in a home-made Openstack cluster, then sent to 30 production instances across 3 EC2 regions.
+Standard Containers are INFRASTRUCTURE-AGNOSTIC: they can be run in any OCI supported infrastructure.
+For example, a standard container can be bundled on a laptop, uploaded to S3, downloaded, run and snapshotted by a build server at Equinix in Virginia, uploaded to 10 staging servers in a home-made OpenStack cluster, then sent to 30 production instances across 3 EC2 regions.
```
Comment (rgardler) [inline principles.md]: It is inappropriate to mention specific infrastructure since someone will always be left out of such examples. Suggest rewording (e.g. "S3" = "cloud storage", OpenStack = "private cloud", "EC2 regions" = "public cloud regions"

Comment (vbatts) [inline principles.md]: good point.

**Comment by duglin** [comment]
@rgardler fixed

**Comment by mrunalp** [comment]
LGTM ping @crosbymichael @philips @LK4D4 @vishh

**Comment by vbatts** [comment]
waiting for one more wording switch

**Comment by duglin** [comment]
done

File: principles.md [hunk 8]
```diff
@@ -3,49 +3,44 @@
...
+Irrespective of their contents, by standardizing the container itself it allowed for a consistent, more streamlined and efficient set of processes to be defined.
+For software Standard Containers offer similar functionality by being the fundamental, standardized, unit of delivery for a software package.
 
 ## 1. Standard operations
 
-Just like shipping containers, Standard Containers define a set of STANDARD OPERATIONS.
-Shipping containers can be lifted, stacked, locked, loaded, unloaded and labelled.
-Similarly, Standard Containers can be created, started, and stopped using standard container tools (what this spec is about); copied and snapshotted using standard filesystem tools; and downloaded and uploaded using standard network tools.
+Standard Containers define a set of STANDARD OPERATIONS.
+They can be created, started, and stopped using standard container tools; copied and snapshotted using standard filesystem tools; and downloaded and uploaded using standard network tools.
```
Comment (vishh) [inline principles.md]: Can we add more clarity here? What is required for hosting these containers to come up with standard network tools? Will OCI define APIs for these purpose?

Comment (wking) [inline principles.md]: On Wed, Oct 07, 2015 at 10:12:46AM -0700, Vish Kannan wrote:
> > +They can be created, started, and stopped using standard
> > container tools; copied and snapshotted using standard filesystem
> > tools; and downloaded and uploaded using standard network tools.
> 
> Can we add more clarity here? What is required for hosting these
> containers to come up with standard network tools? Will OCI define
> APIs for these purpose?

Can we punt on this for now?  I think transfering these files with
tar/Nginx/curl, or tar/netcat, or IPFS, etc., etc., should be fine.
But this has been a hot-button issue in the past, and I'd rather avoid
having discussion about that stall this rewording PR.

**Comment by vbatts** [comment]
cool.

File: principles.md (hunk 8 above)
Comment (duglin) [inline principles.md]: The entire "standard operations" concept is one I'm struggling with and plan on opening a PR on once I get formulate a coherent proposal in my head. So for now I'd prefer if this PR just focused on cleaning up the docs - leaving the hard (potentially controversial) changes for more focused PRs.

Comment (vbatts) [inline principles.md]: yeah. for now, this is just carry over from the prior wording, but this is the fuzziest of terms. Let's clear it up in another PR.
